### ISSUE TITLE
Enumerators for different settings

### ISSUE THREAD

**Issue by Luk164** [SciSharp/Keras.NET#140]
I noticed many settings for models and the like are set using strings, which is really unintuitive and also makes it harder to find available settings. I would suggest a few enumerators or at least constants to be defined to make it easier.

Example:
`model.Compile(optimizer:"sgd", loss:"binary_crossentropy", metrics: new string[] { "accuracy" });`

Would be turned into something like:
`model.Compile(optimizer:keras.models.optimizers.sgd, loss:keras.models.lossFunctions.binary_crossentropy, metrics: new string[] { "accuracy" });`
